### ISSUE TITLE
EPIC: Build the Tassadar LLM-computer system (programs→weights, edge variance, studying)

### ISSUE THREAD

**Issue by AtlantisPleb** [OpenAgentsInc/openagents#5313]
## Context

Tracks the full roadmap from the Tassadar run audit (`docs/tassadar/2026-06-18-tassadar-run-actual-state-and-real-training-gap-audit.md`). Read that doc first: §0 defines "training" for Tassadar as the Percepta **LLM-computer paradigm** (capability is *constructed analytically* — programs compiled into transformer weights — and *verified by exact replay*, not gradient-trained); §2-§3 give the verdict and the gap (the run executes one fixed compiled program `loop_sum_v1` forever, with no variety / dense composable modules / marketplace / pricing of construction); §4 the sequenced top-5 gate; §4b the edge-agents-as-variance-engine design; §4c studying → deep codebase knowledge → Autopilot-as-coder.

This EPIC is the overview and the sequenced checklist of every roadmap issue below.

## The system to build

Programs → compiled weight-modules (construction), verified by exact replay, composed and listed in a marketplace, priced and paid for construction (not just one replay), with the heterogeneous **edge agents** as the variance engine (program authorship, data curation, composition, adversarial verification, demand-priced curation), the **studying → Autopilot-coder** track running in parallel as the near-term product-facing knowledge/data layer, and the **gradient/hybrid ring** entering only at the end as a subordinate frozen-core + learned-interface layer.

## Sequencing

Start with **Track S (studying → Autopilot-coder)** — highest near-term product leverage and the most checkable first data direction. Then interleave **C (construction substrate)** / **V (verification + settlement extensions)** / **E (edge variance engine)**. Finish with **H (hybrid ring)** — gradients enter only there.

## Repo split

`openagents` (this repo) owns execution, replay validation, settlement, projection, the labor market, and the studying/knowledge surfaces. The **construction pipeline lives in the sibling `psionic` repo** (`OpenAgentsInc/psionic`) — the C-track and substrate-completion issues are tracked/coordinated here but **implemented in psionic** (noted in each affected body).

### Track S — Studying → Autopilot-coder (START HERE)
- [x] #5314 — S1: Repo-knowledge substrate v1 — study-packet ingest over the `openagents` repo (current code + commit history + rationale + cross-links; builds on #5284 repo-corpus primitives)
- [x] #5315 — S2: Studied-knowledge graph — cross-link code↔commit↔doc↔issue↔rationale↔invariant (incl. `backroom/` rejected lineage)
- [x] #5316 — S3: Studied-knowledge verification model — derived-trace replay over link/span claims (closes the §4b/§4c data-correctness gap)
- [x] #5317 — S4: StudyBench eval harness over `openagents` — hidden-edit exams + retained-failure fixtures
- [x] #5318 — S5: Autopilot-coder consumes studied knowledge — substrate into the coding-agent context/plan path (the product payoff)
- [x] #5319 — S6: Edge-agent studying contributions as paid work — Artanis labor market + escrow
- [x] #5320 — S7: Generalize studying to external repos (Phase 6 product) — later

### Track C — LLM-computer construction substrate (psionic-heavy)
- [x] #5321 — C1: Run work-unit = real compiled-program corpus (not the single `loop_sum_v1` fixture) — top-5 #1
- [x] #5322 — C2: Complete psionic E4 MILP scheduler (gate-graph→layer/phase assignment) [psionic]
- [x] #5323 — C3: Dense, loadable, digest-pinned weight-module materialization (psionic W1.2) + wire as run artifact — top-5 #2 [psionic + openagents]
- [x] #5324 — C4: Softmax-approximation bounds (psionic W1.4) + Wasm window ladder (W1.1) [psionic]
- [x] #5325 — C5: Module composition/linking (`tassadar_module_linker`) + compiled-weight-module marketplace listing — top-5 #3 [psionic + openagents]

### Track V — Verification + settlement extensions
- [x] #5326 — V1: Extend exact-replay verification + settlement to PAY for construction (not just one fixed replay) — top-5 #4
- [x] #5327 — V2: Module-composition verification (linked module verified as a composition of verified parts)
- [x] #5328 — V3: Data-contribution correctness-verification model (provenance + derived-trace replay + validator review)

### Track E — Edge-agent variance engine
- [x] #5329 — E1: Artanis construction/data work-request directions (program authorship, dataset curation, data-direction proposals)
- [x] #5330 — E2: Demand-priced curation + module-library ranking/dedup
- [x] #5331 — E3: Adversarial-verification market (agents paid to find module-divergence inputs)

### Track H — Hybrid ring (later; gradients enter ONLY here)
- [x] #5332 — H1: Frozen compiled core + learned interface — gradient windows under quarantine/canary/promotion [psionic + openagents]

## Audit doc

`docs/tassadar/2026-06-18-tassadar-run-actual-state-and-real-training-gap-audit.md`

**Comment by AtlantisPleb** [comment]
Track S complete on origin/main.\n\nClosed issues: #5314, #5315, #5316, #5317, #5318, #5319, #5320.\n\nLatest S7 commit: e97dc617b. The studying path now covers dogfood repo ingest/graph/verification/eval/coder context, paid Artanis study contributions, and a yellow refs-only external-repo studying pilot surface with customer/private/marketplace/payout claims blocked.\n\nKnown broader verification state remains unchanged: focused Track S tests pass; full Worker suite still has the unrelated pre-existing failures documented in the S7 closeout.

**Comment by AtlantisPleb** [comment]
Track C complete on `origin/main`.

Closed issues: #5321, #5322, #5323, #5324, #5325.

Latest C5 evidence:
- Psionic linked dense module + fixture: `OpenAgentsInc/psionic@bd1bea24`, `OpenAgentsInc/psionic@4cc09d32`.
- OpenAgents marketplace listing: `OpenAgentsInc/openagents@dc27cf007`.
- Public listing route: `/api/public/tassadar/compiled-module-marketplace`.
- Linked module digest: `cc1403674fc0d38892610d9e9c6c9230075494061f720c45bfa4f7b5a961756a`.
- Verification: psionic linked dense tests, OpenAgents executor tests/typecheck, Worker marketplace tests/typecheck, architecture/freshness checks, and `bun run --cwd apps/openagents.com check:deploy` all passed.

The construction substrate now covers corpus dispatch, MILP scheduling, dense module materialization, softmax/window substrate work, and the first digest-pinned linked-module listing. Remaining roadmap starts Track V: #5326 V1, then #5327 and #5328.

**Comment by AtlantisPleb** [comment]
Roadmap complete.

Final H1/#5332 implementation was pushed in:
- OpenAgentsInc/psionic@8e74fc2d
- OpenAgentsInc/openagents@657cd9270

Completion audit:
- Issues #5314 through #5332 are all CLOSED with stateReason COMPLETED.
- The EPIC checklist has no remaining unchecked items.
- `psionic` and `openagents` are both clean on `main...origin/main`.

Verification for the final H1 work:
- `cargo test -p psionic-tassadar-student`
- `bun run --cwd apps/openagents.com/workers/api test -- src/tassadar-gradient-window-regime.test.ts src/product-promises.test.ts src/tassadar-adversarial-verification-market.test.ts src/artanis-work-directions.test.ts`
- `bun run --cwd apps/openagents.com/workers/api typecheck`
- `bun run --cwd apps/openagents.com check:architecture`
- `bun run --cwd apps/openagents.com check:public-projection-freshness`
- `bun run --cwd apps/openagents.com check:deploy`

No production settlement was armed or run, and no live settlement/deploy behavior was expanded.
